pkgs/build-support/fetchzip/default.nix

```diff
@@ -13,10 +13,19 @@
 , urls ? []
 , extraPostFetch ? ""
 , name ? "source"
+, # A temporary name of the downloaded file. Can be used to set the
+  # proper extension of the file.
+  #
+  # If empty, use the basename of `url' (or of the first element of
+  # `urls').
+  downloadName ? ""
```

Comment:
```suggestion
, # Allows to set the extension for the intermediate downloaded
  # file. This can be used as a hint for the unpackCmdHooks to
  # select an appropriate unpacking tool.
  extension ? null
```